packages/compiler-cli/src/ngtsc/annotations/src/component.ts

```diff
@@ -1431,3 +1442,35 @@ function makeCyclicImportInfo(
       `The ${type} '${name}' is used in the template but importing it would create a cycle: `;
   return makeRelatedInformation(ref.node, message + path);
 }
+
+
+/**
+ * Checks whether a selector is a valid custom element tag name.
+ * Based loosely on https://mothereff.in/custom-element-name.
+ */
+function validateCustomElementSelector(selector: string): string|null {
+  // Avoid flagging components with an attribute selector. This isn't bulletproof since it won't
+  // catch cases like `foo[]bar`, but we don't need it to be. This is mainly to avoid flagging
+  // something like `foo-bar[baz]` incorrectly.
+  if (selector.includes('[') && selector.includes(']')) {
+    return null;
+  }
+
+  if (!selector.includes('-')) {
+    return 'Selector of a ShadowDom-encapsulated component must contain a hyphen.';
```

Comment:
nit: I'd propose rephrasing it a bit to reference the full form (i.e. `ViewEncapsulation.ShadowDom`):
```suggestion
    return 'Selector of a component that uses ViewEncapsulation.ShadowDom must contain a hyphen.';
```